diff --git a/sway/input/seatop_default.c b/sway/input/seatop_default.c
index 55c9159a..08ef1ff9 100644
--- a/sway/input/seatop_default.c
+++ b/sway/input/seatop_default.c
@@ -73,16 +73,17 @@ static enum wlr_edges find_edge(struct sway_container *cont,
 	}
 
 	enum wlr_edges edge = 0;
-	if (cursor->cursor->x < cont->x + cont->border_thickness) {
+	int tolerance = 0;
+	if (cursor->cursor->x < cont->x + cont->border_thickness + tolerance) {
 		edge |= WLR_EDGE_LEFT;
 	}
-	if (cursor->cursor->y < cont->y + cont->border_thickness) {
+	if (cursor->cursor->y < cont->y + cont->border_thickness + tolerance) {
 		edge |= WLR_EDGE_TOP;
 	}
-	if (cursor->cursor->x >= cont->x + cont->width - cont->border_thickness) {
+	if (cursor->cursor->x >= cont->x + cont->width - cont->border_thickness - tolerance) {
 		edge |= WLR_EDGE_RIGHT;
 	}
-	if (cursor->cursor->y >= cont->y + cont->height - cont->border_thickness) {
+	if (cursor->cursor->y >= cont->y + cont->height - cont->border_thickness - tolerance) {
 		edge |= WLR_EDGE_BOTTOM;
 	}
 
